presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ChecksumResult.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.verifier.checksum;
+
+import com.facebook.presto.spi.type.SqlVarbinary;
+import com.google.common.annotations.VisibleForTesting;
+
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Objects.requireNonNull;
+
+public class ChecksumResult
+{
+    private final long rowCount;
+    private final Map<String, Object> checksums;
+
+    @VisibleForTesting
+    ChecksumResult(long rowCount, Map<String, Object> checksums)
+    {
+        for (Object checksum : checksums.values()) {
+            checkArgument(
+                    checksum == null || checksum instanceof Long || checksum instanceof Double || checksum instanceof SqlVarbinary,
+                    "checksum can only be one of (null, Long, Double, SqlVarbinary), found %s",
+                    checksum == null ? "null" : checksum.getClass().getName());
+        }
+        this.rowCount = rowCount;
+        this.checksums = unmodifiableMap(requireNonNull(checksums, "checksums is null"));
```

Comment:
`requireNonNull(checksums)` should go to the top because we would have already accessed it in `checksum.values()`